### PR TITLE
Issue #1043 Allow the configuration of a IntervalBiFunction in Spring Boot2

### PR DESCRIPTION
**Pull request by cosminseceleanu**
Pull request for issue https://github.com/resilience4j/resilience4j/issues/1043

Added support to configure a `IntervalBiFunction` in Spring Boot2

### CODE REVIEW

File: resilience4j-framework-common/src/main/java/io/github/resilience4j/common/retry/configuration/RetryConfigurationProperties.java [hunk 1]
```diff
@@ -243,6 +245,12 @@ private void configureRandomizedWait(InstanceProperties properties, RetryConfig.
         @Nullable
         private Duration waitDuration;
 
+        /*
+         * retry intervalBiFunction class to be used to calculate wait based on exception or result
+         */
+        @Nullable
+        private Class<? extends IntervalBiFunction<Object>> intervalBiFunction;
```
Comment (cosminseceleanu) [inline resilience4j-framework-common/src/main/java/io/github/resilience4j/common/retry/configuration/RetryConfigurationProperties.java:252]: @RobWin Should I throw and exception when both `RetryConfigurationProperties.InstanceProperties.waitDuration` and `RetryConfigurationProperties.InstanceProperties.intervalBiFunction` are used or let `RetryConfig.Builder.build` to do the job?

File: resilience4j-framework-common/src/main/java/io/github/resilience4j/common/retry/configuration/RetryConfigurationProperties.java [hunk 2]
```diff
@@ -18,10 +18,7 @@
 import io.github.resilience4j.common.CommonProperties;
 import io.github.resilience4j.common.CompositeCustomizer;
 import io.github.resilience4j.common.utils.ConfigUtils;
-import io.github.resilience4j.core.ClassUtils;
-import io.github.resilience4j.core.ConfigurationNotFoundException;
-import io.github.resilience4j.core.IntervalFunction;
-import io.github.resilience4j.core.StringUtils;
+import io.github.resilience4j.core.*;
```
Comment (cosminseceleanu) [inline resilience4j-framework-common/src/main/java/io/github/resilience4j/common/retry/configuration/RetryConfigurationProperties.java:21]: Will fix this

File: resilience4j-framework-common/src/main/java/io/github/resilience4j/common/retry/configuration/RetryConfigurationProperties.java (hunk 1 above)
Comment (RobWin) [reply resilience4j-framework-common/src/main/java/io/github/resilience4j/common/retry/configuration/RetryConfigurationProperties.java:252]: Let RetryConfig.Builder.build do the job

**Comment by RobWin** [comment]
Done? Can I review?

**Comment by cosminseceleanu** [comment]
I think so :). Yes, you can review.

**Review by RobWin — APPROVED** [review]

**Pull request merged**